quickwit-config/src/server_config.rs

```diff
@@ -56,27 +98,97 @@ impl IndexerConfig {
     }
 }
 
+impl Default for IndexerConfig {
+    fn default() -> Self {
+        Self {
+            data_dir_path: default_data_dir_path(),
+            rest_listen_address: default_listen_address(),
+            rest_listen_port: Self::default_rest_listen_port(),
+            grpc_listen_address: default_listen_address(),
+            grpc_listen_port: Self::default_grpc_listen_port(),
+            split_store_max_num_bytes: Self::default_split_store_max_num_bytes(),
+            split_store_max_num_splits: Self::default_split_store_max_num_splits(),
+        }
+    }
+}
+
 // TODO: caching
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
 pub struct SearcherConfig {
+    #[serde(default = "default_data_dir_path")]
     pub data_dir_path: PathBuf,
+    #[serde(default = "SearcherConfig::default_host_key_path")]
+    pub host_key_path: PathBuf,
+    #[serde(default = "default_listen_address")]
+    pub rest_listen_address: String,
+    #[serde(default = "SearcherConfig::default_rest_listen_port")]
     pub rest_listen_port: u16,
+    #[serde(default = "default_listen_address")]
+    pub grpc_listen_address: String,
+    #[serde(default = "SearcherConfig::default_grpc_listen_port")]
     pub grpc_listen_port: u16,
+    #[serde(default = "default_listen_address")]
+    pub discovery_listen_address: String,
+    #[serde(default = "SearcherConfig::default_discovery_listen_port")]
     pub discovery_listen_port: u16,
     #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub peer_seeds: Vec<String>,
-    #[serde(default = "default_fast_field_cache_capacity")]
+    #[serde(default = "SearcherConfig::default_fast_field_cache_capacity")]
     pub fast_field_cache_capacity: Byte,
-    #[serde(default = "default_split_footer_cache_capacity")]
+    #[serde(default = "SearcherConfig::default_split_footer_cache_capacity")]
     pub split_footer_cache_capacity: Byte,
 }
 
-fn default_fast_field_cache_capacity() -> Byte {
-    Byte::from_bytes(1_000_000_000) // 1G
-}
+impl SearcherConfig {
+    fn default_host_key_path() -> PathBuf {
+        PathBuf::from("/var/lib/quickwit/host_key")
+    }
+
+    fn default_rest_listen_port() -> u16 {
+        7280
+    }
+
+    fn default_grpc_listen_port() -> u16 {
+        7281
+    }
 
-fn default_split_footer_cache_capacity() -> Byte {
-    Byte::from_bytes(500_000_000) // 500M
+    fn default_discovery_listen_port() -> u16 {
+        7282
```

Comment:
This forces the user that wants to run 2 instances to define these 3 ports.
Opening an issue to see if we can do better.